lib/src/message.dart

```diff
@@ -12,11 +12,12 @@ import 'body.dart';
 import 'headers.dart';
 import 'http_unmodifiable_map.dart';
 
+/// Retrieves the [Body] contained in the [message].
 Body getBody(Message message) => message._body;
 
 /// The default set of headers for a message created with no body and no
 /// explicit headers.
-final _defaultHeaders = new HttpUnmodifiableMap<String>({"content-length": "0"},
+final _defaultHeaders = new HttpUnmodifiableMap<String>({},
```

Comment:
Actually, re-reading [the spec](https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html), it says

> The presence of a message-body in a request is signaled by the inclusion of a Content-Length or Transfer-Encoding header field in the request's message-headers.

So it's probably fine to just not include a `content-length` header for requests that have no entity body, regardless of their method. We'll need to make sure that `dart:io` doesn't force a chunked transfer encoding in those cases, though.